app/models/check_flag_job.rb

```diff
@@ -0,0 +1,51 @@
+class CheckFlagJob < Struct.new(:flag, :token)
+
+  def perform
+    flag.access_token = token
+    flag.resolved?
+  end
+
+  def success(job)
+    flag.job_id = nil
+    flag.save
+  end
+
+  def error(job, exception)
+    @exception = exception
+  end
+
+  def failure(job)
+    flag.job_id = nil
+    flag.status = "failed"
+    flag.save
+  end
+
+  def max_attempts
+    # Retries happen at 5 + n^4 seconds, where n = number of attempts
+    # 12 gives last retry at 6 hours
+    return 12
+  end
+
+  def reschedule_at(time_now, attempts)
+    if (@exception.is_a?(Foursquare2::APIError) && @exception.type == 'rate_limit_exceeded') # Rate limit hit
+
+      # Photo and Tip flags have their own endpoint with their own rate limit
+      if ["PhotoFlag", "TipFlag"].include? flag.type
+        flags_of_type = flag.user.flags.where(:status => "queued").where(:type => flag.type).count
```

Comment:
Flags being checked are in the submitted stage, but we probably want to calculate this based on both queued and submitted. consider replacing the status line with:

:status => ["queued", "submitted"]

(here and on L37)